src/test/java/com/exactpro/junit5/WithByteBufInvocationContext.java

```diff
@@ -13,49 +13,50 @@
 
     private final List<ByteBuf> arguments;
 
-    private StringBuilder displayName;
+    private String methodName;
 
     private WithByteBufMethodContext methodContext;
 
     WithByteBufInvocationContext(String methodName, WithByteBufMethodContext methodContext, List<ByteBuf> arguments) {
-        this.displayName = new StringBuilder(methodName);
+        this.methodName = methodName;
         this.arguments = arguments;
         this.methodContext = methodContext;
     }
 
     @Override
     public String getDisplayName(int invocationIndex) {
-        return formatName(arguments);
+        return formatName(methodName, arguments);
     }
 
     @Override
     public List<Extension> getAdditionalExtensions() {
         return singletonList(new WithByteBufParameterResolver(methodContext, this.arguments));
     }
 
-    private String formatName(Collection<ByteBuf> arguments) {
+    private String formatName(String methodName, Collection<ByteBuf> arguments) {
+        StringBuilder displayName = new StringBuilder(methodName);
         for (ByteBuf buf : arguments) {
-            appendFormattedBuffer(buf);
+            appendFormattedBuffer(displayName, buf);
         }
         return displayName.toString();
     }
 
-    private void appendFormattedBuffer(ByteBuf buf) {
+    private void appendFormattedBuffer(StringBuilder currentName, ByteBuf buf) {
         int readerIndex = buf.readerIndex();
         int readLimit = buf.writerIndex();
-        displayName.append(" [ ");
+        currentName.append(" [ ");
         while (readerIndex < readLimit) {
-            appendHexByte(buf.getByte(readerIndex++));
+            appendHexByte(currentName, buf.getByte(readerIndex++));
         }
-        displayName.append("]");
+        currentName.append("]");
```

Comment:
when you add just one character it's more efficient to add it as char (`']'`) instead of string